### PR TITLE
Main interface features

### PR DESCRIPTION
**Pull request by RefactorForge**
Impelementation of main interface with functions contorling flow of the application, entry main point and some basic test cases.

Kindly asking for review

Thank you

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        validated_config = load_and_validate_config(args.config)
+        container = ClusteringContainer()
+        container.config.from_dict(validated_config.model_dump(mode="json"))
+        io_handler = container.io_handler()
+
+        train_data, train_sample_weight = load_data(
+            io_handler, args.input, args.sample_weight
+        )
+
+        result = execute_clustering(container, train_data, train_sample_weight)
```
Comment (tynaneu) [inline main.py]: execute clustering can also have just the clustering instance (as io_handler) and not take the whole container - you should be consistent in these things

File: app/flow_functions.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        print(config.clustering.algorithm)
+        # Access clustering algorithm details
+        ```
+    """
+    try:
+        with open(config_path, "r") as file:
+            raw_config = yaml.safe_load(file)
+        validated_config = Config(**raw_config)
+        return validated_config
+    except Exception as e:
```
Comment (tynaneu) [inline app/flow_functions.py:72]: except only ValidationError not all exceptions. If there was some other exception raised the reraised exception (with the message "Validation Error" would be misleading.

File: main.py (hunk 1 above)
Comment (RefactorForge) [reply main.py]: Refactored.

File: app/flow_functions.py (hunk 2 above)
Comment (RefactorForge) [reply app/flow_functions.py:72]: Refactored.

**Comment by RefactorForge** [comment]
@tynaneu 
Should be resolved now.
Kindly asking for review.

Thank you

File: main.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        train_data, train_sample_weight = load_data(
+            input_handler, args.input, args.sample_weight
+        )
+
+        result = execute_clustering(
+            clustering_algorithm, train_data, train_sample_weight
+        )
+
+        save_results(output_handler, result, args.output)
+    except Exception as e:
```
Comment (tynaneu) [inline main.py:57]: The function currently uses a generic except Exception block, which suppresses all exceptions, including unexpected ones, making debugging difficult. It should handle specific exceptions where possible and use tools loging stack traces for better debugging. Additionally, re-raising exceptions after logging ensures critical errors aren't silently ignored, maintaining transparency and reliability.

Comment (RefactorForge) [reply main.py:57]: Refactored.

**Comment by RefactorForge** [comment]
@tynaneu 
Should be resolved now.
Kindly asking for review.

Thank you

**Pull request merged**